ts/features/wallet/onboarding/cobadge/screens/start/CoBadgeSingleBankScreen.tsx

```diff
@@ -1,56 +1,73 @@
-  confirmButtonProps
-} from "../../../../../bonus/bonusVacanze/components/buttons/ButtonConfigurations";
+import TosBonusComponent from "../../../../../bonus/bonusVacanze/components/TosBonusComponent";
+import SearchStartScreen from "../../../common/searchBank/SearchStartScreen";
+import { abiListSelector } from "../../../store/abi";
 import { navigateToOnboardingCoBadgeSearchAvailable } from "../../navigation/action";
-import { walletAddCoBadgeCancel } from "../../store/actions";
+import { searchUserCoBadge, walletAddCoBadgeCancel } from "../../store/actions";
 
-type Props = ReturnType<typeof mapDispatchToProps> &
+type OwnProps = {
+  abi: string;
+};
+type Props = OwnProps &
+  ReturnType<typeof mapDispatchToProps> &
   ReturnType<typeof mapStateToProps>;
 
+const tos_url =
+  "https://io.italia.it/app-content/privacy_circuiti_internazionali.html";
+
+const partecipatingBank_url =
+  "https://io.italia.it/cashback/carta-non-abilitata-pagamenti-online";
 /**
  * The initial screen of the co-badge workflow (starting with a specific ABI, eg. from BANCOMAT screen)
  * The user can see the selected bank and can start the search for all the co-badge for the specific bank.
+ * @param _
  * @constructor
- * @param props
  */
-const CoBadgeSingleBankScreen = (props: Props): React.ReactElement => (
-  <BaseScreenComponent
-    goBack={true}
-    headerTitle={I18n.t("wallet.onboarding.coBadge.headerTitle")}
-    contextualHelp={emptyContextualHelp}
-  >
-    <SafeAreaView style={IOStyles.flex}>
-      <Content style={IOStyles.flex}>
-        <H1>CoBadgeSingleBanksScreen</H1>
-      </Content>
-      <FooterWithButtons
-        type={"TwoButtonsInlineThird"}
-        leftButton={cancelButtonProps(props.cancel)}
-        rightButton={confirmButtonProps(props.startSearch)}
-      />
-    </SafeAreaView>
-  </BaseScreenComponent>
-);
+const CoBadgeSingleBankScreen = (props: Props): React.ReactElement => {
+  const { showModal, hideModal } = useContext(LightModalContext);
+
+  const openTosModal = () => {
+    showModal(<TosBonusComponent tos_url={tos_url} onClose={hideModal} />);
+  };
+
+  const openPartecipatingBankModal = () => {
+    showModal(
+      <TosBonusComponent tos_url={partecipatingBank_url} onClose={hideModal} />
+    );
+  };
+
+  const abiInfo: Abi | undefined = props.abiList.find(
+    aI => aI.abi === props.abi
+  );
+  return (
+    <SearchStartScreen
+      methodType={"cobadge"}
+      onCancel={props.onCancel}
+      onSearch={props.searchAccounts}
+      handlePartecipatingBanksModal={openPartecipatingBankModal}
+      handleTosModal={openTosModal}
+      bankName={abiInfo?.name}
+    />
+  );
+};
 
 const mapDispatchToProps = (dispatch: Dispatch) => ({
-  cancel: () => dispatch(walletAddCoBadgeCancel()),
-  startSearch: () => dispatch(navigateToOnboardingCoBadgeSearchAvailable())
+  onCancel: () => dispatch(walletAddCoBadgeCancel()),
+  searchAccounts: (abi?: string) => {
+    dispatch(searchUserCoBadge.request(abi));
```

Comment:
```suggestion
```
The start of the action is already handled in `SearchAvailableCoBadgeScreen`

https://github.com/pagopa/io-app/blob/315b0f44b286d692d096efa954a6cc144628e608/ts/features/wallet/onboarding/cobadge/screens/search/SearchAvailableCoBadgeScreen.tsx#L89